core/src/telemetry/setup.ts

```diff
@@ -0,0 +1,180 @@
+    }
+    if (otelHooks.logRecordProcessors) {
+      logRecordProcessors.push(...otelHooks.logRecordProcessors);
+    }
+  }
+
+  // Try to set up OTel tracing.
+  // If the TracerProvider was already set outside of ADK, this would be a no-op
+  // and results in a warning. In such case we rely on user setup.
+  if (spanProcessors.length > 0) {
+    const newTracerProvider = new NodeTracerProvider({
+      resource: otelResource,
+      spanProcessors: spanProcessors
+    });
+
+    newTracerProvider.register();
+
+    trace.setGlobalTracerProvider(newTracerProvider);
+  }
+
+  // Try to set up OTel metrics.
+  // If the MeterProvider was already set outside of ADK, this would be a no-op
+  // and results in a warning. In such case we rely on user setup.
+  if (metricReaders.length > 0) {
+    const newMeterProvider = new MeterProvider({
+      readers: metricReaders,
+      resource: otelResource,
+    });
+
+    metrics.setGlobalMeterProvider(newMeterProvider);
+  }
+
+  // Try to set up OTel logging.
+  // If the LoggerProvider was already set outside of ADK, this would be a no-op
+  // and results in a warning. In such case we rely on user setup.
+  if (logRecordProcessors.length > 0) {
+    const newLoggerProvider = new LoggerProvider({
+      resource: otelResource,
+      processors: logRecordProcessors,
+    });
+    
+    // logs is experimental, reference to https://open-telemetry.github.io/opentelemetry-js/modules/_opentelemetry_api-logs.html#alpha-software---use-at-your-own-risk
+    logs.setGlobalLoggerProvider(newLoggerProvider);
+  }
+}
+
+/**
+ * Gets the OTel resource with environment variable detection.
+ * 
+ * The resource detection populates resource labels from
+ * environment variables like OTEL_SERVICE_NAME and OTEL_RESOURCE_ATTRIBUTES.
+ * 
+ * @returns A Resource object with detected attributes
+ */
+function getOtelResource(): Resource {
+  return detectResources({
+    detectors: [],
+  });
+}
+
+/**
+ * Gets OTel exporters based on environment variables.
+ * 
+ * @returns OTelHooks containing configured exporters
+ */
+function getOtelExporters(): OTelHooks {
+  const hooks: OTelHooks = {
+    spanProcessors: [],
+    metricReaders: [],
+    logRecordProcessors: [],
+  };
+
+  // Check for trace endpoint
+  if (process.env.OTEL_EXPORTER_OTLP_ENDPOINT || process.env.OTEL_EXPORTER_OTLP_TRACES_ENDPOINT) {
+    hooks.spanProcessors!.push(new BatchSpanProcessor(new OTLPTraceExporter()));
+  }
+
+  // Check for metrics endpoint
+  if (process.env.OTEL_EXPORTER_OTLP_ENDPOINT || process.env.OTEL_EXPORTER_OTLP_METRICS_ENDPOINT) {
+    hooks.metricReaders!.push(new PeriodicExportingMetricReader({
+      exporter: new OTLPMetricExporter(),
+    }));
+  }
+
+  // Check for logs endpoint
+  if (process.env.OTEL_EXPORTER_OTLP_ENDPOINT || process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
+    hooks.logRecordProcessors!.push(new BatchLogRecordProcessor(new OTLPLogExporter()));
+  }
+
+  return hooks;
+}
```

Comment:
we can split it into 2 functions and make it simpler and similar to the `google_cloud.ts` file:

```suggestion
function getOtelExportersConfig(): OtelExportersConfig {
  return {
    enableTracing: process.env.OTEL_EXPORTER_OTLP_ENDPOINT || process.env.OTEL_EXPORTER_OTLP_TRACES_ENDPOINT,
    enableMetrics: process.env.OTEL_EXPORTER_OTLP_ENDPOINT || process.env.OTEL_EXPORTER_OTLP_METRICS_ENDPOINT,
    enableLogging: process.env.OTEL_EXPORTER_OTLP_ENDPOINT || process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT,
  };
}

function getOtelExporters(config = getOtelExportersConfig()): OTelHooks {
  const { enableTracing, enableMetrics, enableLogging } = config;

  return  {
    spanProcessors: enableTracing ? [new BatchSpanProcessor(new OTLPTraceExporter())] : [],
    metricReaders: enableMetrics ? [new PeriodicExportingMetricReader({ exporter: new OTLPMetricExporter() })] : [],
    logRecordProcessors: enableLogging ? [new BatchLogRecordProcessor(new OTLPLogExporter())] : [],
  };
}
```